src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfaceProgress.cs

```diff
@@ -60,19 +70,56 @@ class ConsoleHostUserInterface : System.Management.Automation.Host.PSHostUserInt
                 _pendingProgress = new PendingProgress();
             }
 
-            _pendingProgress.Update(sourceId, record);
-
             if (_progPane == null)
             {
-                // This is the first time we've received a progress record. Create a progress pane, then show it.
+                // This is the first time we've received a progress record
+                // Create a progress pane
+                // Set up a update flag
+                // Create a timer for updating the flag
 
                 _progPane = new ProgressPane(this);
+
+                if (_progPaneUpdateTimer == null && _progPane != null)
+                {
+                    // Show a progress pane at the first time we've received a progress record
+                    progPaneUpdateFlag = true;
+
+                    // Invoke the timer only once to exclude overlaps
+                    // The timer will be restarted in 'ProgressPaneUpdateTimerElapsed'
+                    _progPaneUpdateTimer = new Timer( new TimerCallback(ProgressPaneUpdateTimerElapsed), null, UpdateTimerThreshold, Timeout.Infinite);
+                }
+            }
+
+            if (progPaneUpdateFlag || record.RecordType == ProgressRecordType.Completed)
+            {
+                // Update the progress pane only when the timer set up the update flag or WriteProgress is completed.
+                // As a result, we do not block WriteProgress and whole script and eliminate unnecessary console locks and updates.
+                _pendingProgress.Update(sourceId, record);
+                _progPane.Show(_pendingProgress);
+                progPaneUpdateFlag = false;
```

Comment:
It's possible the timer fires right after `_progPane.Show` and set `progPaneUpdateFlag` to true, but then it gets immediately set to false here. This will cause the `Show` to not be called for at least another 200 ms, which make the showing becomes uneven. 

I gave it some thoughts and came up with my version of `HandleIncomingProgressRecord` based on your change:
```
        private
        void
        HandleIncomingProgressRecord(Int64 sourceId, ProgressRecord record)
        {
            Dbg.Assert(record != null, "record should not be null");

            if (_pendingProgress == null)
            {
                Dbg.Assert(_progPane == null, "If there is no data struct, there shouldn't be a pane, either.");

                _pendingProgress = new PendingProgress();
            }

            _pendingProgress.Update(sourceId, record);

            if (_progPane == null)
            {
                // This is the first time we've received a progress record
                // Create a progress pane
                // Set up a update flag
                // Create a timer for updating the flag

                _progPane = new ProgressPane(this);

                if (_progPaneUpdateTimer == null)
                {
                    // Show a progress pane at the first time we've received a progress record
                    progPaneUpdateFlag = 1;

                    // Invoke the timer only once to exclude overlaps
                    // The timer will be restarted in 'ProgressPaneUpdateTimerElapsed'
                    _progPaneUpdateTimer = new Timer(new TimerCallback(ProgressPaneUpdateTimerElapsed), null, UpdateTimerThreshold, UpdateTimerThreshold);
                }
            }

            if (Interlocked.CompareExchange(ref progPaneUpdateFlag, 0, 1) == 1 || 
                record.RecordType == ProgressRecordType.Completed)
            {
                // Update the progress pane only when the timer set up the update flag or WriteProgress is completed.
                // As a result, we do not block WriteProgress and whole script and eliminate unnecessary console locks and updates.
                _progPane.Show(_pendingProgress);
            }
        }
```